File: leetcode/135/candy.py
class Solution:
    def candy(self, ratings: List[int]) -> int:
        lst = []
        for i in range(len(ratings)):
            if len(lst) > 0 and ratings[i-1] < ratings[i]:
                lst.append(lst[-1]+1)
            else:
                lst.append(1)

        for i in range(len(ratings)-1, 0, -1):
            if ratings[i-1] > ratings[i] and lst[i-1] <= lst[i]:
                lst[i-1] = lst[i]+1

        return sum(lst)


        

File: leetcode/135/candy.swift
class Solution {
    func candy(_ ratings: [Int]) -> Int {
        var lst : [Int] = [1]

        for i in 1..<ratings.count{
            if ratings[i-1] < ratings[i]{
                lst.append(lst[i-1]+1)
            }
            else{
                lst.append(1)
            }
        }

        for i in stride(from: ratings.count-1, through: 1, by: -1){
            if ratings[i-1] > ratings[i] && lst[i-1] <= lst[i]{
                lst[i-1] = lst[i]+1
            }
        }
        return lst.reduce(0,+)
    }
}
